src/metpy/calc/tools.py

```diff
@@ -781,6 +781,41 @@ def take(indexer):
     return take
 
 
+@exporter.export
+def find_local_extrema(var, nsize, extrema):
+    r"""Find the local extreme (max/min) values of a 2D array.
+
+    Parameters
+    ----------
+    var : `xarray.DataArray`
+        The variable to locate the local extrema using the nearest method
+        from the maximum_filter or minimum_filter from the scipy.ndimage module.
+    nsize : int
+        The minimum number of grid points between each local extrema.
+    extrema: str
+        The value 'max' for local maxima or 'min' for local minima.
+
+    Returns
+    -------
+    var_extrema: `xarray.DataArray`
+        The values of the local extrema with other values as NaNs
+
```

Comment:
Is requiring xarray buying us anything here? (It looks like `maximum_filter`, etc. work fine when given a `DataArray`. Also, what about an API where instead of getting back the data field with values masked, just getting back the mask directly? Something like (which also addresses a CodeQL complaint):

```python
    from scipy.ndimage import maximum_filter, minimum_filter

    if extrema == 'max':
        extreme_val = maximum_filter(var.values, nsize, mode='nearest')
    elif extrema == 'min':
        extreme_val = minimum_filter(var.values, nsize, mode='nearest')
    else:
        raise ValueError(f'Invalid value for "extrema": {extrema}. Valid options are "max" or "min".')

    return var == extreme_val
```

Or alternatively we could return arrays of x,y indices of the locations of the maxima? A benefit of both of these alternatives is that they lend themselves to more use cases and avoid dealing with units.